pkg/server/registry/apigroups/acorn/apps/validator.go

```diff
@@ -279,6 +283,28 @@ func (s *Validator) validateName(app *apiv1.App) error {
 	return nil
 }
 
+func (s *Validator) validateTargetServiceName(app *apiv1.App) (string, error) {
+	workloads := app.Status.AppSpec.Containers
+
+	for _, p := range app.Spec.Publish {
+		if p.TargetServiceName != "" && len(workloads) > 0 {
+			if _, ok := workloads[p.TargetServiceName]; !ok {
+				return p.TargetServiceName, fmt.Errorf("valid container name(s) are [%s] received [%q]", containerNames(workloads), p.TargetServiceName)
+			}
+			continue
+		}
+	}
+	return "", nil
+}
+
+func containerNames(containers map[string]v1.Container) string {
```

Comment:
This function looks the same as in parse.go. See if it makes sense to have it in common utility and access it from both the places.